katdal/test/test_chunkstore_s3.py

```diff
@@ -73,28 +75,24 @@ def setup(self):
                                                endpoint_url=url,
                                                aws_access_key_id='blah',
                                                aws_secret_access_key='blah')
-                self.store = S3ChunkStore(client)
+                cls.store = S3ChunkStore(client)
         except Exception:
-            self.teardown()
+            cls.teardown_class()
             raise
 
-    def teardown(self):
-        if self.fakes3:
-            self.fakes3.terminate()
-            self.fakes3.wait()
-        shutil.rmtree(self.tempdir)
+    @classmethod
+    def teardown_class(cls):
+        if cls.fakes3:
+            cls.fakes3.terminate()
+            cls.fakes3.wait()
+        shutil.rmtree(cls.tempdir)
 
     def array_name(self, path):
         bucket = 'katdal-unittest'
         return self.store.join(bucket, path)
 
-
-class TestDudS3ChunkStore(object):
-    """Tests that don't need an S3 connection, only a 'dud' store."""
-
-    def setup(self):
-        if not botocore:
-            raise SkipTest('S3 botocore dependency not installed')
-
     def test_store_unavailable(self):
-        assert_raises(StoreUnavailable, S3ChunkStore.from_url, 'http://i.nvalid/')
+        # Drastically reduce the default botocore timeout of nearly 7 seconds
+        assert_raises(StoreUnavailable, S3ChunkStore.from_url,
+                      'http://i.nvalid/',
```

Comment:
Rather use a `.invalid` address. There is an RFC that guarantees that `.invalid` will never be a top-level domain, whereas this could theoretically be a valid address in future.

Reply:
I did not know that... Here's the [RFC](https://tools.ietf.org/html/rfc6761). Done.